scope-docs/metadata-schema/variable_details.yaml

```diff
@@ -0,0 +1,204 @@
+schema_version: "0.1"
+schema_date: "2025-01-01"
+description: "Core variable details schema including active templateVariable development"
+
+variable_details_schema:
+  title: "CCHSFlow Variable Details Configuration"
+  description: "Schema for variable_details.csv, defining value-level attributes, recoding logic, and categorical value labels for variables."
+  version: "0.1"
+  id_column_name: "file_row_id"
+  expected_column_order:
+    - "file_row_id"
+    - "variable"
+    - "templateVariable"    # Active field - currently in development
+    - "dummyVariable"
+    - "typeEnd"
+    - "typeStart"
+    - "databaseStart"       # camelCase convention
+    - "variableStart"       # camelCase convention
+    - "variableStartLabel"  # camelCase convention
+    - "numValidCat"
+    - "recEnd"
+    - "catLabel"
+    - "catLabelLong"
+    - "units"
+    - "recStart"
+    - "variableStartShortLabel"  # camelCase convention
+    - "notes"
+
+  fields:
+    - name: "file_row_id"
+      title: "File Row Identifier"
+      description: "Unique identifier for the row within this CSV file. Generated using format: detail_{variable_name}_{sequence}"
+      type: "string"
+      constraints:
+        required: true
+        unique: true
+        pattern: "^detail_[a-zA-Z0-9_.]+_[0-9]{3}$"  # Semantic ID pattern with sequence
+    - name: "variable"
+      title: "Variable Name"
+      description: "Canonical name of the variable this detail row pertains to. Foreign key to variables.csv."
+      type: "string"
+      constraints:
+        required: true
+    - name: "templateVariable"
+      title: "Template Variable Indicator"
+      description: "Indicates if this variable follows a template pattern or references another template variable"
+      type: "string"
+      constraints:
+        required: false
+        # Permissive validation - allowing both "Yes"/"No" and template variable names
+    - name: "dummyVariable"
+      title: "Dummy Variable Indicator"
+      description: "Indicates if this row defines a dummy variable created during recoding (eg, for a category of a categorical variable)."
+      type: "string"
+      constraints:
+        required: false
+    - name: "typeEnd"
+      title: "Target Data Type"
+      description: "The data type of the variable *after* recoding or as its final representation (eg, categorical, numeric)."
+      type: "string"
+      constraints:
+        required: false
+    - name: "typeStart"
+      title: "Source Data Type"
+      description: "The data type of the variable *before* recoding or in its original form."
+      type: "string"
+      constraints:
+        required: false
+    - name: "databaseStart"
+      title: "Original Database Name"
+      description: "Name of the original database or data source for this variable detail."
+      type: "string"
+      constraints:
+        required: false
+    - name: "variableStart"
```

Comment:
Should follow the validation rules in the variables sheet? Some way to share the two rules?

Reply:
Yes. Checked that patterns are the same and updated.